osidb/models.py

```diff
@@ -2896,14 +2906,18 @@ class PackageVersions(PolymorphicModel):
     class Meta:
         """define meta"""
 
-        verbose_name = "Package Versions"
+        verbose_name = "Old Package Versions"
 
     def validate(self, *args, **kwargs):
         """validate package versions model"""
         super().clean_fields(*args, **kwargs)
 
 
 class CVEv5PackageVersions(PackageVersions):
+    """
+    This model is deprecated. Use Package instead. Delete when it's clear
+    that Package works correctly.
+    """
```

Comment:
Failing forwards might imply manual resync, right?

The old models ale left there temporarily. I guess we can remove them in a few weeks after a demo. This is so that we still have the old correct data in case there's a bug or unintended interaction with other changes that's discovered only in stage or production (as was the case a few weeks ago, discovered on the day of a demo). Once the old models are deleted, we can't revert the data migration from old to new models. Since fixed_in is in tens of thousands of flaws, a manual resync is too lengthy an endeavor. I propose to let the old models in there until the new API is demoed (performed read&write on stage and read-only on production) and then delete them in a separate PR.

Reply:
You added `meta_attr["fixed_in"] = self.flaw_bug["fixed_in"]` in BZImport in this PR, and this solves the data backup issue. Each flaw will have its original value here, so you can delete the old unused models. You just need to do the migrations for the change in `meta_attr`.

Also, I am not sure if we can delete model's fields without deprecating them first via `deprecate_field_custom` to keep N-1 compatibility. Or can we do it if these fields are unused in serializers/APIs? WDYT, @Elkasitu?

Reply:
> and this solves the data backup issue

No. Already explained why, for [example here](https://github.com/RedHatProductSecurity/osidb/blob/7bd1b89b7045e2f557cd198b2dd9c6895a4f63cf/apps/bbsync/query.py#L394).

> You just need to do the migrations for the change in meta_attr.

No, the data is not saved in the database in osidb master. Bzimport parses that data but throws the original string away. Only the original string contains the original separators. You also raised it [here, with a similar answer](https://github.com/RedHatProductSecurity/osidb/pull/307#discussion_r1323363923).

> Also, I am not sure if we can delete model's fields without deprecating them first via deprecate_field_custom to keep N-1 compatibility.

I don't understand. Please search for `@extend_schema_serializer(deprecate_fields=["status"])` in the PR's diff.

Reply:
Whether you deprecate the field in the model or API is different. See the discussion in #PR267

Reply:
From the linked DEVELOP.md file:

> For the DB we need to ensure that OSIDB is N-1 compatible meaning that the app of version N can work with the DB of version N-1 and vice-versa.

This quality is met with this PR. Note that existing models are not touched at all (except for the `related_name`).

> For the API we need to ensure that the breaking changes are introduced only in major releases. The standard procedure is as follows.

This PR adds new API endpoints and doesn't change the existing field (except for marking one field as deprecated).